src/redisearch_rs/rlookup/src/lookup.rs

```diff
@@ -267,16 +267,12 @@ impl<'a> DerefMut for RLookupKey<'a> {
 // This means you may NEVER EVER hand out a `&mut CStr` EVER.
 impl<'a> RLookupKey<'a> {
     /// Constructs a new `RLookupKey` using the provided `CStr` and flags.
-    ///
-    /// If the [`RLookupKeyFlag::NameAlloc`] is given, then the provided `CStr` will be cloned into
-    /// a new allocation that is owned by this key. If the flag is *not* provided the key
-    /// will simply borrow the provided string.
-    pub fn new(name: &'a CStr, flags: RLookupKeyFlags) -> Self {
-        let name = if flags.contains(RLookupKeyFlag::NameAlloc) {
-            Cow::Owned(name.to_owned())
-        } else {
-            Cow::Borrowed(name)
-        };
+    pub fn new(name: impl Into<Cow<'a, CStr>>, flags: RLookupKeyFlags) -> Self {
+        debug_assert!(
+            !flags.contains(RLookupKeyFlag::NameAlloc),
+            "the NameAlloc flag only exists for C-compatability and must not be passed to RLookupKey::new"
```

Comment:
Will we need to re-implement this in the future? Or what's the expected behavior when we need to clone the name?